DEVELOPER_GUIDE.md

```diff
@@ -27,6 +30,61 @@ changes to Fluent Bit.
     - [Valgrind](#valgrind)
   - [Need more help?](#need-more-help)
 
+## Development Environment
+
+Development environments allow to work without any dependency locally.
+
+### Devcontainer
+
+A Development Container (or Dev Container for short) allows you to use a container as a full-featured development environment.
+
+Thanks to [Visual Studio Code integration](https://code.visualstudio.com/docs/devcontainers/containers), devcontainers are managed easily.
+
+#### Prerequirements
+- [docker](https://www.docker.com)
+- [visual studio code](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) [optional]
+- [devcontainer](https://github.com/devcontainers/cli) [optional]
+
+#### Using without optional tools
+
+```shell
+docker run \
+    --name devcontainer-fluent-bit \
+    --volume $PWD/:/workspaces/fluent-bit \
+    --tty \
+    --detach \
+    fluent/fluent-bit:latest-debug
+docker exec devcontainer-fluent-bit addgroup --system --gid 1000 fluent-bit
```

Comment:
Maybe link out to documentation on using your local UID and GID? It may not be 1000.

Should it be in a setup script to catch all this into a one-liner.

Reply:
I think the option should not be one-liner. Because each step should appear in documentation. If a developer dont use an extension, he/she should known how to run container.

we can provide uid and gid with environment variables on linux. It is not required [for windows or mac.](https://jtreminio.com/blog/running-docker-containers-as-current-host-user/#macos-and-windows-users)